sdk-base/src/main/java/com/mapbox/maps/threading/AnimationThreadController.kt

```diff
@@ -0,0 +1,112 @@
+package com.mapbox.maps.threading
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Looper
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * Experimental controller singleton allowing to use background thread for all Mapbox animations.
+ * Most likely should not be used directly, calling [useBackgroundThread] should do all the work.
+ *
+ * By default all Mapbox animations are running on Android Main thread.
+ *
+ * Following Mapbox plugins are updated to make use of background animation thread:
+ *  - plugin-locationcomponent
+ *  - plugin-animation
+ *  - plugin-viewport
+ *
+ *  Switching between [useBackgroundThread] and [useMainThread] could be done at any point in runtime.
+ *
+ *  Using this class is extremely unsafe and error prone, it should be used in very specific use-cases!
+ *  Even when it's essential to use it (for example Android Auto use-case on specific devices when locking the device)
+ *  it would be safer to use Android Main thread as much as possible and switch to background thread
+ *  only when it's needed.
+ */
+@MapboxExperimental
+object AnimationThreadController {
```

Comment:
@kiryldz will you add tests for this?

i find objects easier to unit test if they have a internal class that is essentially identical

``` kotlin
object AnimationThreadController {
  private val delegate = AnimationThreadControllerDelegate()
  
  fun useBackgroundThread() = delegate.useBackgroundThread()

...
}
```

Reply:
first step will make sure that all existing tests are passing, tests for background will be added later separately. However thanks for the architectural tip, I will look into it